packages/jest-util/src/ErrorWithStack.ts

```diff
@@ -10,9 +10,15 @@ export default class ErrorWithStack extends Error {
     message: string | undefined,
     callsite: (...args: Array<any>) => unknown,
   ) {
+    // Ensure we have a large stack length so we get full details.
+    const stackLimit = Error.stackTraceLimit;
+    Error.stackTraceLimit = Math.max(100, stackLimit || 10);
```

Comment:
this may have an adverse impact on performance as we use this for every single defined test and many assertions. Should we pass in an argument instead allowing handles to have a longer stack, but keeping current behavior as-is?

Reply:
👍 that makes sense to me. Can do.